qhana_plugin_runner/api/util.py

```diff
@@ -242,3 +242,51 @@ def deserialize(
 
 
 # TODO add plugin URL field + UI generator functions
+class PluginUrl(ma.fields.URL):
+    """Extension of the URL field that can handle plugin URLs"""
+
+    def __init__(
+        self,
+        *,
+        relative: bool = False,
+        schemes: Optional[types.StrSequenceOrSet] = None,
+        require_tld: bool = True,
+        plugin_tags: Optional[Union[Sequence[str], str]] = None,
+        plugin_name: Optional[str] = None,
+        plugin_version: Optional[str] = None,
+        required: bool = False,
+        allow_none: bool = False,
+        **kwargs,
+    ):
+        super().__init__(
+            relative=relative,
+            schemes=schemes,
+            require_tld=require_tld,
+            allow_none=allow_none,
+            required=required,
+            **kwargs,
+        )
+        if plugin_tags:
+            self.metadata["plugin_tags"] = plugin_tags
+        if plugin_name:
+            self.metadata["plugin_name"] = plugin_name
+        if plugin_version:
+            self.metadata["plugin_version"] = plugin_version
+        self.validators[0] = FileUrlValidator(
```

Comment:
The FileUrlValidator also validates file:// urls which should not work in this case. Either adapt (and rename) the file URL validator to enable/disable file:// urls on demand or create a separate validator.